drivers/mipi_dsi/dsi_mcux_2l.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2023, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ /* Based on dsi_mcux.c, which is (c) 2022 NXP */
+
+#define DT_DRV_COMPAT nxp_mipi_dsi_2l
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/mipi_dsi.h>
+#include <fsl_mipi_dsi.h>
+#include <fsl_clock.h>
+#include <zephyr/logging/log.h>
+
+#include <soc.h>
+
+LOG_MODULE_REGISTER(dsi_mcux_host, CONFIG_MIPI_DSI_LOG_LEVEL);
+
+struct mcux_mipi_dsi_config {
+	MIPI_DSI_HOST_Type *base;
+	dsi_dpi_config_t dpi_config;
+	bool auto_insert_eotp;
+	const struct device *bit_clk_dev;
+	clock_control_subsys_t bit_clk_subsys;
+	const struct device *esc_clk_dev;
+	clock_control_subsys_t esc_clk_subsys;
+	const struct device *pixel_clk_dev;
+	clock_control_subsys_t pixel_clk_subsys;
+	uint32_t dphy_ref_freq;
+};
+
+struct display_mcux_mipi_dsi_data {
+	const struct device *dev;
```

Comment:
Looks like it is not used anywhere, also not initialized.

Reply:
You're right, not sure how this got missed. Thanks for catching it, I've remove this structure